.github/workflows/build-test.yml

```diff
@@ -21,11 +21,25 @@ jobs:
                 cloud_provider: [ 'AWS', 'AZURE', 'GCP' ]
         steps:
             - uses: actions/checkout@v1
+            - name: Restore cached deps
+              id: cache-restore-deps
+              uses: actions/cache/restore@v3
```

Comment:
there is version 4 available (also for save action) https://github.com/actions/cache

Reply:
I had issues with version 4 while testing. Maybe we can upgrade later on.